iriscasttools/iriscasttools/utils.py

```diff
@@ -0,0 +1,224 @@
+"""
+Provides utility functions to collect energy usage information
+"""
+import subprocess
+import time
+import os
+from pathlib import Path
+import re
+from typing import Dict
+import logging
+import datetime
+
+logger = logging.getLogger(__name__)
+
+
+def retry(
+    retry_on: (Exception,) = None,
+    retries: int = 3,
+    delay: int = 3,
+    backoff: int = 2,
+    retry_logger: logging.Logger = None,
```

Comment:
We can capture the logger from the module scope here to remove boilerplate later on:

```
retry_logger: logging.Logger = logger
```